groups/reconcile.go

```diff
@@ -0,0 +1,397 @@
+package main
+
+import (
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/admin/directory/v1"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/groupssettings/v1"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	// the email id for the bot/service account
+	BotID string `yaml:"bot-id"`
+
+	// the file with the authentication information
+	TokenFile string `yaml:"token-file,omitempty"`
+
+	// the file with the groups/members information
+	GroupsFile string `yaml:"groups-file,omitempty"`
+
+	// If true, don't make any mutating API calls
+	DryRun bool
+}
+
+type GroupsConfig struct {
+	// List of google groups
+	Groups []GoogleGroup `yaml:"groups,omitempty"`
+}
+
+type GoogleGroup struct {
+	EmailId     string `yaml:"email-id"`
+	Description string `yaml:"description"`
+
+	// List of members in the google group
+	Members []string `yaml:"members,omitempty"`
+}
+
+func Usage() {
+	fmt.Fprintf(os.Stderr, `
+Usage: %s [-config <config-yaml-file>] [-dry-run]
+Command line flags override config values.
+`, os.Args[0])
+	flag.PrintDefaults()
+}
+
+var config Config
+var groupsConfig GroupsConfig
+
+func main() {
+	configFilePath := flag.String("config", "config.yaml", "the config file in yaml format")
+	dryRun := flag.Bool("dry-run", false, "do not push anything to google groups")
```

Comment:
Would prefer to keep default / exploratory runs safe, make mutation explicit.  Would prefer you flip this to a flag like `--confirm` that defaults to false and must explicitly be set to true.  We do this in test-infra for things like [label_sync](https://github.com/kubernetes/test-infra/tree/master/label_sync) and [peribolos](https://github.com/kubernetes/test-infra/tree/master/prow/cmd/peribolos#settings)

Reply:
Done. switch over to `--confirm` style